python/ray/util/iter.py

```diff
@@ -527,8 +527,9 @@ def base_iterator(timeout=None):
                             batch_ms)] = actor
                         for item in batch:
                             yield item
-                    except StopIteration:
-                        pass
+                    except StopIteration as ex:
+                        if isinstance(ex, ForceIteratorStopIteration):
```

Comment:
@Edilmo if you raise exception here,  then it would shut down with retryable error and restart the AI Engine. I don't think it makes sense in cases like all workers are in the "pending state".

Reply:
More importantly, I think I'm more interested in the scope you think about on what the  architect (especially execution plan) looks like to handle normal/exception sim cases and the associated expected behaviors, instead of fixing them one-by-one. What do you think? I think perhaps a design doc for people to review is a starting point. (cc: @bimalkmehta)